preupg/application.py

```diff
@@ -623,14 +614,15 @@ def run(self):
 
         if not self.conf.scan and not self.conf.contents and \
                 not self.conf.list_rules:
-            ret_val = self._check_available_contents()
-            if int(ret_val) != 0:
-                return ret_val
+            self.conf.scan = self._get_default_module_set()
+            if not self.conf.scan:
+                return ReturnValues.SCENARIO
 
         if self.conf.list_rules:
-            ret_val = self._check_available_contents()
-            if int(ret_val) != 0:
-                return ret_val
+            if not self.conf.scan:
```

Comment:
The line 622 is basically the only line needed for the fix. The other changes were done to make the code a bit more sane.